File: Models/PlanStatus.swift
import Foundation

enum PlanStatus: String, Codable, CaseIterable, Identifiable {
    case geplant
    case inArbeit = "in-arbeit"
    case fertig

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .geplant: return "Geplant"
        case .inArbeit: return "In Arbeit"
        case .fertig: return "Fertig"
        }
    }
    
    var symbolName: String {
        switch self {
        case .geplant: return "circle.fill"
        case .inArbeit: return "clock.fill"
        case .fertig: return "checkmark.circle.fill"
        }
    }
}

File: Windows-App/Unterrichtsplaner/Models/PlanStatus.cs
namespace Unterrichtsplaner.Models;

public enum PlanStatus
{
    Geplant,
    InArbeit,
    Fertig
}

public static class PlanStatusExtensions
{
    public static string DisplayName(this PlanStatus s) => s switch
    {
        PlanStatus.Geplant  => "Geplant",
        PlanStatus.InArbeit => "In Arbeit",
        PlanStatus.Fertig   => "Fertig",
        _ => s.ToString()
    };

    public static string Icon(this PlanStatus s) => s switch
    {
        PlanStatus.Geplant  => "\u25cb",
        PlanStatus.InArbeit => "\u25b6",
        PlanStatus.Fertig   => "\u2713",
        _ => "\u25cb"
    };
}
